src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -301,12 +302,17 @@ private static void ReportUseSiteDiagnostics(Symbol symbol, DiagnosticBag diagno
             }
         }
 
-        private static MethodSymbol GetRequiredMethod(TypeSymbol type, string methodName, DiagnosticBag diagnostics)
+        private static MethodSymbol GetRequiredMethod(TypeSymbol type, string methodName, DiagnosticBag diagnostics, Location location = null)
```

Comment:
What is a use-site diagnostic?  How do you inject a "use-site-diagnostic" into a document that has no real "uses" of the synthesized main?

---
In reply to: [111510678](https://github.com/dotnet/roslyn/pull/18472#discussion_r111510678) [](ancestors = 111510678)